datafusion/sqllogictest/bin/sqllogictests.rs

```diff
@@ -491,9 +497,7 @@ impl TestFile {
     }
 }
 
-fn read_test_files<'a>(
-    options: &'a Options,
-) -> Result<Box<dyn Iterator<Item = TestFile> + 'a>> {
+fn read_test_files(options: &Options) -> Result<Vec<TestFile>> {
```

Comment:
This function was already basically returning a `Vec` so there didn't seem to be any need to have the iterator indirection